### PR TITLE
Allow to create feeder bays in a bus-breaker voltage level

### PR DESCRIPTION
**Pull request by miovd**
Signed-off-by: VEDELAGO MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature: allow to use modifications to create feeder bays in a bus-breaker voltage level.

**What is the current behavior?** *(You can also link to an open issue here)*
Modifications to create feeder bays only work in node-breaker voltage levels.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
Yes, deprecation
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ModificationReports.java [hunk 1]
```diff
@@ -19,49 +19,133 @@ final class ModificationReports {
...
+                .withSeverity(TypedValue.INFO_SEVERITY)
+                .build());
+    }
+
+    static void newCouplingDeviceAddedReport(Reporter reporter, String voltageLevelId, String bbsId1, String bbsId2, int nbOpenDisconnectors) {
+        reporter.report(Report.builder()
+                .withKey("newCouplingDeviceAdded")
+                .withDefaultMessage("New coupling device was created on voltage level ${voltageLevelId}. It connects busbar sections ${bbsId1} and ${bbsId2} with closed disconnectors" +
+                        "and ${nbOpenDisconnectors} were created on parallel busbar sections.")
+                .withValue(voltageLevelId, voltageLevelId)
```
Comment (flo-dup) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ModificationReports.java:90]:
```suggestion
                .withValue(VOLTAGE_LEVEL_ID, voltageLevelId)
```

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ModificationReports.java [hunk 2]
```diff
@@ -99,85 +173,97 @@ static void noConnectablePositionExtension(Reporter reporter, VoltageLevel volta
...
 
-    static void voltageLevelRemovingEquipmentsLeftReport(Reporter reporter, String vlId) {
+    static void removeFeederBayBusbarSectionReport(Reporter reporter, String busbarSectionConnectableId) {
         reporter.report(Report.builder()
-                .withKey("voltageLevelRemovingEquipmentsLeft")
-                .withDefaultMessage("Voltage level ${vlId} still contains equipments")
-                .withValue("vlId", vlId)
-                .withSeverity(TypedValue.WARN_SEVERITY)
+                .withKey("removeBayBusbarSectionConnectable")
+                .withDefaultMessage("Cannot remove feeder bay for connectable ${busbarSectionConnectableId}, as it is a busbarSection")
```
Comment (flo-dup) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ModificationReports.java:254]:
```suggestion
                .withDefaultMessage("Cannot remove feeder bay for connectable ${connectableId}, as it is a busbarSection")
```

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java [hunk 3]
```diff
@@ -57,24 +59,54 @@ protected AbstractCreateConnectableFeederBays(int... sides) {
...
+        }
+        Connectable<?> connectable = add();
+        if (!checkNetworks(connectable, network, reporter, throwException)) {
+            return;
+        }
+
+        LOGGER.info("New connectable {} of type {} created", connectable.getId(), connectable.getType());
+        createdConnectable(reporter, connectable);
+
+        createExtensionAndTopology(connectable, network, reporter);
```
Comment (flo-dup) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java:73]: it's clearer that way!

**Review by flo-dup — CHANGES_REQUESTED** [review]

**Review by flo-dup — APPROVED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java [hunk 4]
```diff
@@ -83,14 +115,20 @@ public void apply(Network network, boolean throwException, ComputationManager co
...
+    }
 
+    private void createExtensionAndTopology(Connectable<?> connectable, Network network, Reporter reporter) {
+        String connectableId = connectable.getId();
         boolean createConnectablePosition = false;
         ConnectablePositionAdder<?> connectablePositionAdder = connectable.newExtension(ConnectablePositionAdder.class);
         for (int side : sides) {
             VoltageLevel voltageLevel = getVoltageLevel(side, connectable);
+            if (voltageLevel.getTopologyKind() != TopologyKind.NODE_BREAKER) {
+                continue; // no extension nor switches created in bus-breaker topology
```
Comment (miovd) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AbstractCreateConnectableFeederBays.java:130]: Add log message if position not null and are ignored

**Review by annetill — APPROVED** [review]

**Review by flo-dup — APPROVED** [review]

**Pull request merged**